lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -158,10 +97,179 @@ def strip_var_names(cubes):
+        to auxiliary coordinate to be associated with a time dimension.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube requiring addition of the specified coordinates as
+                auxiliary coordinates.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube where the the requested coordinates have been added to the
+                cube as auxiliary coordinates and associated with the desired
+                master coordinate.
+
+        Raises:
+            ValueError: If the master coordinate is not present on the cube.
+        """
+        coordinates = self.coords_to_associate
+        if coordinates is None:
+            coordinates = []
+
+        # If the master_coord is not a dimension coordinate, then the other
+        # coordinates cannot be associated with it.
+        if len(cube.coords(self.master_coord, dim_coords=True)) > 0:
+            for coord in coordinates:
+                if cube.coords(coord):
+                    temp_coord = cube.coord(coord)
+                    cube.remove_coord(coord)
+                    temp_aux_coord = (
+                        build_coordinate(temp_coord.points,
+                                         bounds=temp_coord.bounds,
+                                         coord_type=AuxCoord,
+                                         template_coord=temp_coord))
+                    coord_names = [
+                        coord.standard_name for coord in cube.dim_coords]
+                    cube.add_aux_coord(
+                        temp_aux_coord,
+                        data_dims=coord_names.index(self.master_coord))
+        return cube
+
+    @staticmethod
+    def _slice_over_coordinate(cubes, coord_to_slice_over):
+        """
+        Function slices over the requested coordinate and promotes the sliced
+        coordinate into a one-point dimension to help concatenation.
+
+        Args:
+            cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cubes to be concatenated.
+            coord_to_slice_over (list):
+                Coordinate to slice over.
+
+        Returns:
+            sliced_by_coord_cubelist (iris.cube.CubeList):
+                CubeList containing sliced cubes.
+        """
+        sliced_by_coord_cubelist = iris.cube.CubeList([])
+        if isinstance(cubes, iris.cube.Cube):
+            cubes = iris.cube.CubeList([cubes])
+
+        for cube in cubes:
+            if cube.coords(coord_to_slice_over):
+                for coord_slice in cube.slices_over(coord_to_slice_over):
+                    coord_slice = iris.util.new_axis(
+                        coord_slice, coord_to_slice_over)
+                    sliced_by_coord_cubelist.append(coord_slice)
+            else:
+                sliced_by_coord_cubelist.append(cube)
+
+        return sliced_by_coord_cubelist
+
+    def process(self, cubes_in):
+        """
+        Concatenate cubes
```

Comment:
I'm a fan of brevity, but this seems a little extreme. Is there nothing interesting to say about process at all? If not, I guess a full stop would be something at least 😄